src/test/java/org/logstash/beats/BeatsParserTest.java

```diff
@@ -50,14 +53,25 @@ public void testEncodingDecodingFields() {
         assertMessages(this.batch, decodedBatch);
     }
 
-
     @Test
-    public void  testCompressedEncodingDecodingFields() {
+    public void testCompressedEncodingDecodingFields() {
         this.batch.setProtocol(Protocol.VERSION_1);
         Batch decodedBatch = decodeCompressedBatch();
         assertMessages(this.batch, decodedBatch);
     }
 
+    @Test
+    public void testShouldNotCrashOnGarbageData() {
```

Comment:
I've changed the expectation and the exception is now propagated down stream in the BeatsHandler, which should have been in the first place.